core/arch/interp.c

```diff
@@ -3506,16 +3512,22 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             }
             if (!d_r_is_avx512_code_in_use()) {
                 if (ZMM_ENABLED()) {
-                    if (instr_get_prefix_flag(bb->instr, PREFIX_EVEX)) {
-                        /* For AVX-512 detection in bb builder, we're checking only for
-                         * the prefix flag, which for example can be set by decode_cti. In
-                         * client_process_bb, post-client instructions are checked with
-                         * instr_may_write_zmm_register.
-                         */
-                        LOG(THREAD, LOG_INTERP, 2, "Detected AVX-512 code in use\n");
-                        d_r_set_avx512_code_in_use(true);
-                        proc_set_num_simd_saved(MCXT_NUM_SIMD_SLOTS);
+#    ifdef CLIENT_INTERFACE
+                    if (!TEST(INSTR_CLEANCALL_SIMD_CONTEXT, bb->instr->flags)) {
```

Comment:
I don't see how this flag could be set here.  I think the flag is not needed: see above.

Reply:
See above, done.